src/main/java/com/redhat/devtools/lsp4ij/features/formatting/LSPClientSideOnTypeFormattingTypedHandler.java

```diff
@@ -50,120 +51,108 @@ public Result charTyped(char charTyped,
+                        onTypeFormattingSettings.formatOnStatementTerminator |= formattingFeature.isFormatOnStatementTerminator(file);
+                        FormattingScope formatOnStatementTerminatorScope = formattingFeature.getFormatOnStatementTerminatorScope(file);
+                        if (formatOnStatementTerminatorScope.compareTo(onTypeFormattingSettings.formatOnStatementTerminatorScope) > 0) {
+                            onTypeFormattingSettings.formatOnStatementTerminatorScope = formatOnStatementTerminatorScope;
+                        }
+                        onTypeFormattingSettings.formatOnStatementTerminatorCharacters += formattingFeature.getFormatOnStatementTerminatorCharacters(file);
+
+                        onTypeFormattingSettings.formatOnCompletionTrigger |= formattingFeature.isFormatOnCompletionTrigger(file);
+                        onTypeFormattingSettings.formatOnCompletionTriggerCharacters += formattingFeature.getFormatOnCompletionTriggerCharacters(file);
                     }
                 }
-            }
+        );
+        boolean rangeFormattingSupported = rangeFormattingSupportedRef.get();
 
-            // Statement terminators
-            if (formattingFeature.isFormatOnStatementTerminator(file) &&
+        // Close braces
+        if (onTypeFormattingSettings.formatOnCloseBrace &&
                 // Make sure the formatter supports formatting of the configured scope
-                ((formattingFeature.getFormatOnStatementTerminatorScope(file) == FormattingScope.FILE) || rangeFormattingSupported)) {
-                String formatOnStatementTerminatorCharacters = formattingFeature.getFormatOnStatementTerminatorCharacters(file);
-                if (StringUtil.isNotEmpty(formatOnStatementTerminatorCharacters) &&
-                    (formatOnStatementTerminatorCharacters.indexOf(charTyped) > -1)) {
-                    return handleStatementTerminatorTyped(
+                ((onTypeFormattingSettings.formatOnCloseBraceScope == FormattingScope.FILE) || rangeFormattingSupported)) {
+            Map.Entry<Character, Character> bracePair = ContainerUtil.find(
+                    LSPIJEditorUtils.getBracePairs(file).entrySet(),
+                    entry -> entry.getValue() == charTyped
+            );
+            if (bracePair != null) {
+                Character openBraceChar = bracePair.getKey();
+                Character closeBraceChar = bracePair.getValue();
+                if (StringUtil.isEmpty(onTypeFormattingSettings.formatOnCloseBraceCharacters) ||
+                        (onTypeFormattingSettings.formatOnCloseBraceCharacters.indexOf(closeBraceChar) > -1)) {
+                    return handleCloseBraceTyped(
                             project,
                             editor,
                             file,
-                            formattingFeature,
-                            charTyped
+                            onTypeFormattingSettings.formatOnCloseBraceScope,
```

Comment:
We were previously passing `formattingFeature` here just to get the formatting scope, so now we just pass the formatting scope directly.